### PR TITLE
Implement Role-Based Access Control (RBAC) using CASL

### PR DESCRIPTION
**Pull request by HanifZafari**
# Implement Role-Based Access Control (RBAC)

## Description

This pull request implements Role-Based Access Control (RBAC) system using the CASL library (@casl/ability and @casl/react). The changes introduce fine-grained permissions based on user roles.

### Key Changes:

1. **Dependency Updates**: 
   - Updated package dependencies for @casl/ability and @casl/react

2. **CASL Implementation**:
   - Added ability definitions for different user roles in abilities
   - Updated the auth store to set abilities based on user's role whenever a team is changed. 

3. **UI Updates**:
   - Created a new PermissionButton component for handling permissions and rendering buttons based on user's ability
   - Used the PermissionButton to replace normal buttons wherever RBAC required.

## Testing Instructions

Please test the following scenarios:

1. Log in with different user roles and verify that UI elements (especially buttons) are correctly shown/hidden based on permissions.
2. Attempt to perform actions with different user roles to ensure that permissions are correctly enforced.
 
## Additional Notes

- Abilities can be found in ./frontend/util/abilities.ts, this is where all the abilities for each user role is defined.

### CODE REVIEW

File: frontend/src/screens/AddRecruitmentRoundPage.tsx
Comment (JesseCr00) [inline frontend/src/screens/AddRecruitmentRoundPage.tsx:1]: This page needs a permission button on Submit too

File: frontend/src/screens/AdminAcceptPage.tsx
Comment (JesseCr00) [inline frontend/src/screens/AdminAcceptPage.tsx:1]: Maybe don't need permissions on here. It's fine to keep it, just make sure all roles have this permission. (Might limit in future)

File: frontend/src/util/abilities.ts
Comment (JesseCr00) [inline frontend/src/util/abilities.ts:1]: Do owners and admins have all the permissions that team lead does?

File: frontend/src/screens/FeedbackNote.tsx
Comment (JesseCr00) [inline frontend/src/screens/FeedbackNote.tsx:1]: For the permission buttons here, again can Admin and Owner do it as well?

File: frontend/src/screens/ViewRecruitmentRoundPage.tsx [hunk 1]
```diff
@@ -246,13 +256,17 @@ const ViewRecruitmentRoundPage = () => {
           </Grid>
 
           <Box>
-            <Button
+            <PermissionButton
```
Comment (JesseCr00) [inline frontend/src/screens/ViewRecruitmentRoundPage.tsx:259]: I think let them open the modal, but dont let them edit or save.

**Review by JesseCr00 — CHANGES_REQUESTED** [review]
Looks good, just need a few minor changes.

**Review by JesseCr00 — APPROVED** [review]

**Review by Rahul-Saxena-1 — APPROVED** [review]

**Pull request merged**